benchmarks/bench_text_vectorizers.py

```diff
@@ -0,0 +1,73 @@
+"""
+
+To run this benchmark, you will need,
+
+ * scikit-learn
+ * pandas
+ * memory_profiler
+ * psutil (optional, but recommended)
+
+"""
+
+from __future__ import print_function
+
+import timeit
+import itertools
+
+import numpy as np
+import pandas as pd
+from memory_profiler import memory_usage
+
+from sklearn.datasets import fetch_20newsgroups
+from sklearn.feature_extraction.text import (CountVectorizer, TfidfVectorizer,
+                                             HashingVectorizer)
+
+
+def run_vectorizer(Vectorizer, X, **params):
+    def f():
+        vect = Vectorizer(**params)
+        vect.fit_transform(X)
+    return f
+
+
+text = fetch_20newsgroups(subset='train').data
+
+print("="*80 + '\n#' + "    Text vectorizers benchmark" + '\n' + '='*80 + '\n')
+print("Using a subset of the 20 newsrgoups dataset ({} documents)."
+      .format(len(text)))
+print("This benchmarks runs in ~20 min ...")
+
+res = []
+
+# Wrap the result of iteritools.product with tqdm to get a progress bar
```

Comment:
That was more intended as a justification why the `iteritools.product` was used there, and to prevent somebody from refactoring it into a double `for` loop which would be, after all, simpler and more readable. Since the run time is quite long (~20min) I felt it's useful to have some indication of the the progress, even though I didn't want to add `tqdm` as a dependency (or do some manual progress indicator with `print`)..